lasagne/init.py

```diff
@@ -25,6 +65,22 @@ def sample(self, shape):
 
 
 class Uniform(Initializer):
+    """Sample initial weights from the uniform distribution
+
+    Parameters are sampled from U(a, b).
+
+    Parameters
+    ----------
+    range : float or tuple
+        When std is None then range determines a, b. If range is a float the
+        weights are sampled from U(-range, range). If range is a tuple the
+        weights are sampled from U(range[0], range[1]).
+    std : float or None
+        If std is a float then the weights are sampled from
+        U(mean - np.sqrt(3) * std, mean + np.sqrt(3) * std).
+    mean : float
+        see std for description.
+    """
```

Comment:
@benanne: Reading this, what about changing the functionality like this:
```
range : float or tuple
    If a float, weights are sampled from U(mean - range, mean + range).
    If a tuple, weights are sampled from U(range[0], range[1]).
std : float or None
    If a float, overrides `range` with `np.sqrt(3) * std`. This results in sampling
    from a uniform distribution with a standard deviation of `std`.
mean : float
    Mean of the sampled uniform distribution if `range` or `std` are given as a
    scalar, ignored otherwise.
```
The difference is that `mean` would be included if `range` is set to a scalar.